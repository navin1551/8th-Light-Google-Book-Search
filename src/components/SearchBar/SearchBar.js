import React from "react";
import SearchInput from "../SearchInput/SearchInput";
import "./SearchBar.css";

export default class SearchBar extends React.Component {
  render() {
    return (
      <div>
        <div className="Search-bar-area">
          <SearchInput updateBooks={this.props.updateBooks} />
        </div>
      </div>
    );
  }
}
