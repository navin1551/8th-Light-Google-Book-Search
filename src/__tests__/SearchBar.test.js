import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import SearchBar from "../components/SearchBar/SearchBar";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <SearchBar />
    </BrowserRouter>,
    div
  );
});

it("renders SearchBar correctly", () => {
  render(
    <BrowserRouter>
      <SearchBar />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SearchBar />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
