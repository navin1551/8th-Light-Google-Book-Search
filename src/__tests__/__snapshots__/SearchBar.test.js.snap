// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="search-bar-area"
>
  <div>
    <div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="search-input-area"
        >
          <label
            htmlFor="Search"
          >
            Search:
          </label>
          <input
            id="user-search"
            name="Search"
            onChange={[Function]}
            placeholder=""
            type="text"
          />
          <button
            id="search-button"
          >
            Search
          </button>
          <a
            href="/reading-list"
            id="reading-list-link"
            onClick={[Function]}
          >
            My Reading List
          </a>
        </div>
      </form>
      <div
        className="no-value-message-area"
      >
        
      </div>
    </div>
  </div>
</div>
`;
